buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -2229,20 +2248,25 @@ public void run() {
         latch.await(10, TimeUnit.SECONDS);
         barrier.await(5, TimeUnit.SECONDS);
         assertNull(cause.get());
+        buffer.release();
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void readByteThrowsIndexOutOfBoundsException() {
-        final ByteBuf buffer = releaseLater(newBuffer(8));
-        buffer.writeByte(0);
-        assertEquals((byte) 0, buffer.readByte());
-        buffer.readByte();
+        final ByteBuf buffer = newBuffer(8);
+        try {
+            buffer.writeByte(0);
+            assertEquals((byte) 0, buffer.readByte());
+            buffer.readByte();
+        } finally {
```

Comment:
@Scottmitch let us not do it in a finally block for now then... I think we should do a follow up then to either change all to do it in a finally block or change all tests to not do it in a finally block. Agree ?